test/config/ytt/core/overlay-cluster-local-override.yaml

```diff
@@ -0,0 +1,16 @@
+#@ load("@ytt:overlay", "overlay")
+#@ load("@ytt:yaml", "yaml")
+#@ load("@ytt:data", "data")
+
+#@ def rename_cluster_domain(old, new):
```

Comment:
Why would you use `ytt` over `kustomize` or `helm`? 

These files are hard to reason about what they will turn into.

Reply:
I looked at kustomize it is super restrictive and if you have N configurable things with M options you'll need NxM profiles to maintain.

re: helm - how do you think helm would help?  I prefer to be able to post-process the output yamls from `./hack/generate-yamls.sh`.

Reply:
it looks like `ytt` enables you to make a program or function in the comments of your yaml which seems strange at first look.

Reply:
yeah that's essentially what it is - with the language being starlark

You could pull functions out of yaml into `*.star` files ie.
https://carvel.dev/ytt/#example:example-load-custom-library-module